### PR TITLE
refactor: consolidate and harden Anthropic prompt-cache checkpoint handling

### PR DESCRIPTION
**Pull request by dgageot**
`chat.Message.CacheControl` (Anthropic prompt-cache checkpoints) was handled in seven places with implicit cross-layer contracts, which had already produced two production bugs: markers surviving compaction and exceeding Anthropic's 4-breakpoint limit, and a panic in `extractSystemBlocks` on empty flagged messages. This PR consolidates the handling to three logical owners and makes the known failure modes structurally impossible.

On the deduplication side, the two hand-rolled 5-case cache-control switches are replaced with the SDK's `GetCacheControl` union accessor (verified to cover every block kind the converters produce, on both standard and Beta unions — this test already caught a regression when `anthropic-sdk-go` v1.57→v1.58 landed on main mid-branch). The shared message-breakpoint budget helper is extracted, and the entire mechanism — budget ledger, system-block marking, message-tail breakpoints, system-block extraction — is consolidated into `pkg/model/provider/anthropic/cache_control.go`.

On the robustness side, system cache breakpoints are now capped so a request can never exceed Anthropic's 4-breakpoint limit regardless of upstream marks; excess marks are dropped and debug-logged, and the worst case is pinned by test. `session.CompactionInput` clears `Cost`/`CacheControl` on its own copies, removing the compactor's mirrored clearing loops. `session.AddMessage` strips `CacheControl` at ingestion, enforcing the documented contract that transcripts never carry marks — guarding against API clients that echo marks back. `chat.Message.CacheControl` now documents its full contract, including why the field must stay JSON-serialized (marks must survive `before_llm_call` hook rewrite round-trips; out-of-band boundary indices would be silently misplaced by hook message inserts/deletes).

Behavior changes only affect inputs that previously caused API rejections or panics. All other paths are byte-identical, confirmed by e2e cassette replay.

### CODE REVIEW

**Review by Sayt-0 — APPROVED** [review]

File: pkg/model/provider/anthropic/cache_control.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		} else if txt := strings.TrimSpace(msg.Content); txt != "" {
+			// Trim system-message content: YAML literal blocks (instruction: |) always
+			// append a trailing newline, and we don't want that in API payloads.
+			systemBlocks = append(systemBlocks, anthropic.TextBlockParam{
+				Text: txt,
+			})
+		}
+
+		if msg.CacheControl && len(systemBlocks) > 0 {
+			marked = markSystemBlockCacheControl(&systemBlocks[len(systemBlocks)-1], marked)
```
Comment (docker-agent) [inline pkg/model/provider/anthropic/cache_control.go:79]: **[low] extractSystemBlocks: CacheControl=true system message with all-whitespace MultiContent silently mismarks the previous message's last block**

When a system message has `CacheControl=true` **and** `MultiContent` composed entirely of whitespace-only text parts, it contributes zero blocks to `systemBlocks`. The guard `len(systemBlocks) > 0` is satisfied by earlier messages, so `markSystemBlockCacheControl` fires on the *previous* message's last block instead of the intended boundary — silently attributing the cache checkpoint to the wrong location.

```go
// Problematic path: msg contributes no blocks but still triggers mark
if msg.CacheControl && len(systemBlocks) > 0 {
    marked = markSystemBlockCacheControl(&systemBlocks[len(systemBlocks)-1], marked)
}
```

A minimal guard that checks whether this message contributed any blocks would prevent the misattribution:

```suggestion
		blocksBeforeThisMsg := len(systemBlocks)

		if len(msg.MultiContent) > 0 {
			for _, part := range msg.MultiContent {
				if part.Type == chat.MessagePartTypeText {
					if txt := strings.TrimSpace(part.Text); txt != "" {
						systemBlocks = append(systemBlocks, anthropic.TextBlockParam{Text: txt})
					}
				}
			}
		} else if txt := strings.TrimSpace(msg.Content); txt != "" {
			// Trim system-message content: YAML literal blocks (instruction: |) always
			// append a trailing newline, and we don't want that in API payloads.
			systemBlocks = append(systemBlocks, anthropic.TextBlockParam{
				Text: txt,
			})
		}

		if msg.CacheControl && len(systemBlocks) > blocksBeforeThisMsg {
			marked = markSystemBlockCacheControl(&systemBlocks[len(systemBlocks)-1], marked)
		}
```

In practice this combination (CacheControl=true + fully whitespace MultiContent) is unlikely because such a system message carries no useful content. The checkpoint would still be consumed (burning one of the 2 system budget slots) on a block that has nothing to do with the intended message boundary.

| Confidence | Score |
| :--: | :--: |
| 🟡 moderate | 62/100 |

**Review by docker-agent — COMMENTED** [review]
### Assessment: 🟢 APPROVE

This refactor cleanly consolidates the Anthropic prompt-cache checkpoint handling. The budget arithmetic is correct (tool + system + message ≤ 4 in all combinations), the `AddMessage` ingestion strip and `CompactionInput` clearing are correctly implemented, and the `GetCacheControl()` accessor swap is sound (verified by the SDK coverage test that pins all block kinds). One low-confidence edge case in `extractSystemBlocks` is noted inline.
